asset/cloud-client/quickstart_searchallresources_test.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import uuid
+
+import backoff
+from google.api_core.exceptions import NotFound
+from google.cloud import bigquery
+import pytest
+
+import quickstart_searchallresources
+
+PROJECT = os.environ['GOOGLE_CLOUD_PROJECT']
+DATASET = 'dataset_{}'.format(uuid.uuid4().hex)
+
+
+@pytest.fixture(scope='module')
+def bigquery_client():
+    yield bigquery.Client()
+
+
+@pytest.fixture(scope='module')
+def asset_dataset(bigquery_client):
+    dataset = bigquery_client.create_dataset(DATASET)
+
+    yield DATASET
+
+    try:
+        bigquery_client.delete_dataset(dataset)
+    except NotFound as e:
+        print('Failed to delete dataset {}'.format(DATASET))
+        raise e
+
+
+def test_search_all_resources(asset_dataset, capsys):
+    scope = "projects/{}".format(PROJECT)
+    query = "name:{}".format(DATASET)
+
+    # The test will fail before the dataset creation event goes to CAI.
+    @backoff.on_exception(
+        backoff.expo, (AssertionError), max_time=30
+    )
```

Comment:
Is there a way to tell if this dataset is searchable as part of the fixture? I would much prefer a check at the fixture level to determine if the resource is searchable before yielding it. If that can only be accomplished by searching it, it's fine to leave this as is (but please make sure to include either `max_time` or `max_tries` so it doesn't timeout forever).

Reply:
I don't think it's possible unless we do sleep() for a few seconds.